PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ["account.journal"]
-    _inherit = ["account.payment"]
-    _inherit = ['account.payment.method']
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_check_printing/models/account_journal.py
+++ b/addons/account_check_printing/models/account_journal.py
@@ -7,7 +7,7 @@ from odoo.exceptions import ValidationError
 
 
 class AccountJournal(models.Model):
-    _inherit = ["account.journal"]
+    _inherit = "account.journal"
 
     def _default_outbound_payment_methods(self):
         res = super()._default_outbound_payment_methods()

--- a/addons/account_check_printing/models/account_payment.py
+++ b/addons/account_check_printing/models/account_payment.py
@@ -10,7 +10,7 @@ INV_LINES_PER_STUB = 9
 
 
 class AccountPayment(models.Model):
-    _inherit = ["account.payment"]
+    _inherit = "account.payment"
 
     check_amount_in_words = fields.Char(
         string="Amount in Words",

--- a/addons/account_check_printing/models/account_payment_method.py
+++ b/addons/account_check_printing/models/account_payment_method.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class AccountPaymentMethod(models.Model):
-    _inherit = ['account.payment.method']
+    _inherit = 'account.payment.method'
 
     @api.model
     def _get_payment_method_information(self):

--- a/addons/account_check_printing/models/res_company.py
+++ b/addons/account_check_printing/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import models, fields
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     # This field needs to be overridden with `selection_add` in the modules which intends to add report layouts.
     # The xmlID of all the report actions which are actually Check Layouts has to be kept as key of the selection.

--- a/addons/account_check_printing/models/res_config_settings.py
+++ b/addons/account_check_printing/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     account_check_printing_layout = fields.Selection(
         related='company_id.account_check_printing_layout',
